src/controllers/controller.js

```diff
@@ -0,0 +1,139 @@
+  const templateVariables = {
+    region: selection.region,
+    regionArray: dataManager.generateRegionList(),
+    constellationArray: dataManager.constellationFilter(selection.region),
+  };
+  renderTemplate(res, "constellation", templateVariables);
+}
+
+/**
+ * Gets the Materials to render to the template.
+ * @param req
+ * @param res
+ */
+function getMaterials(req, res) {
+  const selection = buildSelection(req);
+  const templateVariables = {
+    region: selection.region,
+    constellation: selection.constellation,
+    regionArray: dataManager.generateRegionList(),
+    constellationArray: dataManager.constellationFilter(selection.region),
+    materialsArray: dataManager.generateMaterialList(),
+  };
+  renderTemplate(res, "material", templateVariables);
+}
+
+/**
+ * Gets the Richness to render to the template. Uses Cookies to remember user previous selection.
+ * @param req
+ * @param res
+ */
+function getRichness(req, res) {
+  const selection = buildSelection(req);
+  const templateVariables = {
+    region: selection.region,
+    constellation: selection.constellation,
+    material: selection.material,
+    regionArray: dataManager.generateRegionList(),
+    constellationArray: dataManager.constellationFilter(selection.region),
+    materialsArray: dataManager.generateMaterialList(),
+    richness: cookie.getCookies(req),
+  };
+  renderTemplate(res, "richness", templateVariables);
+}
+
+/**
+ * Get initial results, creates newly selected richness cookies.
+ * @param req
+ * @param res
+ */
+function getResults(req, res) {
+  cookie.createRichnessCookie(req, res);
+  const richnessArray = Object.getOwnPropertyNames(req.body);
+  buildResults(req, res, richnessArray);
+}
+
+
+/**
+ * Builds initial results and rebuilds resubmitted Data.
+ * @param req
+ * @param res
+ * @param {option? : array} richnessArray
+ */
+function buildResults(req, res, richnessArray) {
+  const selection = buildSelection(req, richnessArray);
+  const results = dataManager.generateResults(selection);
+  results.sort((a, b) => b["Output"] - a["Output"]);
+  const templateVariables = {
+    region: selection.region,
+    constellation: selection.constellation,
+    material: selection.material,
+    regionArray: dataManager.generateRegionList(),
+    constellationArray: dataManager.constellationFilter(selection.region),
+    materialsArray: dataManager.generateMaterialList(),
+    richness: cookie.checkRichness(req, richnessArray),
+    selectedResults: results,
+  };
+  renderTemplate(res, "result-builder", templateVariables);
+}
+
+/**
+ * Construct an object of users Selections.
+ * @param req
+ * @param {option? : Array} richness optional.
+ * @returns {object} of selection(s).
+ */
+function buildSelection(req, richness) {
+  const material = req.body.material ? req.body.material : req.query.material;
+  const region = req.body.region ? req.body.region : req.query.region;
+  const constellation = req.body.region
```

Comment:
This is a bit hard to read, the nested conditionals. Also, that formatting style is not really a good idea. Generally people put ternaries on one line `foo ? bar : baz`

Reply:
Changed to be more simple and easier to read.